Dockerfile

```diff
@@ -0,0 +1,20 @@
+# Use -e ECHO_BOT_TOKEN="Bot exampletoken" as an environment variable when running
+FROM ubuntu:20.04 AS build
+ENV DEBIAN_FRONTEND=noninteractive
+RUN apt update && apt install -y \
+    libboost-all-dev \
+    cmake \
+    build-essential \
+    libssl-dev \
+    libcrypto++-dev \
+    libcurl4-openssl-dev \
+&& rm -rf /var/lib/apt/lists/*
+WORKDIR /echo-bot
```

Comment:
Could you make this an `echo-bot` configurable so it works if users change the project name? This is a template repository.

Reply:
I'm not familiar with template repositories, how do you make this change?

Reply:
Try `RUN export PROJECT=$(grep project CMakeLists.txt | cut -d "(" -f2 | cut -d ")" -f1)` then use `$PROJECT`

Reply:
Overall I think this is a nice concept for a template repository, but I think this point is a bit more complicated. To my knowledge the  `RUN export PROJECT=...` only sets it as an environment variable in the intermediary Ubuntu image. You cannot use `WORKDIR /${PROJECT}` in the Dockerfile itself then as this value is not known outside the intermediary Ubuntu image. Another issue is somehow copying this over to the alpine image.

One way around it is to use ARG and define possible arguments for `docker build` but that costs the end user more time to use than actually changing the file.